### DISCUSSION TITLE
[Balance] Discuss existing MEs here

### DISCUSSION THREAD

**Discussion by Tempo-anon** [Despair-Games/poketernity#1166] [Balance]
### Is your feature request related to a problem? Please describe.

Ideally we would probably want completely new MEs in the game. Use this card to give your thoughts on them, in terms of aesthetics, gameplay balance, how fun/exciting they are to see, as well as how they're implemented under the hood. Implementation wise, we should go for more extendable MEs rather than really bespoke ones that reinvent the wheel.

### Describe the Feature

Here is a list of MEs:

### Common
* Berries Abound
* Department Store Sale
* FIeld Trip
* Fiery Fallout
* Lost at Sea
* Part Timer
* Teleporting Hijinks
* GTS
* Mysterious Chest
* Vitamin dealer
* Uncommon Breed

### Great
* Absolute avarice
* Offer you can't refuse
* Dancing lessons
* Delibirdy
* Fun and games
* Mysterious challengers
* Slumbering Snorlax
* Bug type superfan
* Safari zone
* The strong stuff

### Ultra
* Clowning around
* Expert Pokemon breeder
* Pokemon salesman
* Training session
* Trash to treasure

### Rogue
* A trainer's test
* Dark deal
* The winstrates
* Weird dream

**Comment by Tempo-anon** [comment]
### Berries abound ⭐ ⭐ ⭐ 
Flavor: A strong Pokemon guarding berries makes sense from a lore point of view and also blends together some gameplay elements together. It definitely works
Gameplay: It feels a bit underwhelming compared to other MEs, but I think it works fine for being something a little more unique than a randomly generated opponent. I just think that berries as a whole are pretty underwhelming rewards even if they get a bit more utility from other MEs
Code: Nothing too egregious here, and an okay enough framework for extending into different opponents, different rewards, and different options to bypass the battle

### Department Store Sale ⭐ ⭐ ⭐ ⭐ ⭐ 
Flavor: a giant shopping store is a recurring location in all the mainline games so this fits right in.
Gameplay: An option for specifically choosing a reward category (TM, vitamin, pokeball, x item) is a fantastic way of giving players more choices and rewarding good decisions
Code: Seems pretty simple and really scalable for other events with different reward options


### FIeld Trip ⭐ ⭐ ⭐ ⭐ 
Flavor: Pokemon schools are a common occurrence and this is a cute interaction
Gameplay: Not every ME needs to be some flashy banger. This is a very simple introduction to MEs as well as beginner players while also being a check for button mashers to make sure they aren't asleep. The rewards are simple but should be satisfying. Should be kept as an early game (<50) ME though
Code: Also not too bad to be a springboard for building more quiz like scenarios

### Fiery Fallout ⭐ ⭐ 
Flavor: Having to endure a firestorm is a neat idea but I wonder if having it always be Volcarona 
Gameplay: I have a few gripes with this ME. The combat option always being Volcarona instead of other possible fire Pokemon makes this less unique than the other MEs that feature tough fights. The hunkering down option seems too potentially punishing (changing a random ability to heatproof), and the final option has a pretty lenient requirement (fire type or fire resistance ability). I think the balancing on this definitely needs work
Code: Actually pretty clean and scalable

### Lost at Sea :star: :star: :star: :star:
Flavor: I think every player knows this feeling so capturing it in an ME is a fantastic idea
Gameplay: I quite like this. It's a simple team check that results in some HP penalty if the player doesn't meet it, but the requirement is generous enough (having a Pokemon that can learn surf/fly)
Code: Pretty clean and scalable

### Part Timer :star: :star: :star: :star:
Flavor: A way to have a Pokemon doing work in exchange for money fits very well into the worldbuilding of Pokemon
Gameplay: It's simple but I like it. A quick team stat check for stats in exchange for money (which has improved value due to MEs) is pretty good in lieu of a normal encounter. The "penalty" of reducing PP is also pretty good since it touches on a mechanic that is not too commonly interacted with in the game.
Code: Not too bad but reading a lot of ME code makes me think that a lot of stuff can be moved to helper functions, especially when a choice is made

### Teleporting Hijinks :star: :star:
Flavor: A way to jump to a random biome is something players really want and is quite a fun gameplay mechanic that could use further exploration. There is also an option to bypass the money requirement which is pretty cute
Gameplay: I love the idea of a random biome teleport but this teleport only has space, fairy cave, lab, island, wasteland, and dojo which is sort of ?_? to me. Additionally, I don't know if forcing a boss encounter is really necessary. I feel like this is one of those MEs that has a lot of potential but the way it's currently implemented could use some tweaking
Code: Some minor copying/pasting overall okay

### GTS :star: :star: :star: :star:
Flavor: The GTS and its equivalents are some of the most fun in single player Pokemon games because it gets to both simulate the experience of a trade as well as getting a random Pokemon
Gameplay: Getting a random Pokemon or a choice of a new Pokemon is very fun and exciting. Also providing metaprogression through unlocks makes this even more desirable. Egg move slots should be part of the metaprogression unlocks though.
Code: Helper functions would help this a lot but this could and should be expanded for more trading events

### Mysterious Chest :star: :star: :star:
Flavor: A "mimic" in the Pokerogue sense, I think this works
Gameplay: Perhaps the very first ME created, it's a simple risk vs reward decision. Would prefer to not have the same boss with hardcoded moveset each time but I think this ME is fine.
Code: Seeing some more duplicated code

### Vitamin dealer :star: :star: :star: :star:
Flavor: I think we all know what the shady vitamin dealer is a stand-in for but I think it works okay with enough plausible deniability
Gameplay: I think this is a good ME in that it is one of the MEs that really makes money a more valuable resource during a run. Getting permanent boosts is always a great feeling and being able to convert your money to extra stats is great. Being able to randomize a nature is also interesting but it doesn't update the dex which may cause some confusion to players. Things that give the player new abilities/natures/moves should be consistent in whether or not they actually provide permanent metaprogression
Code: Not too terrible, can potentially be used for other permanent stat gain events

Uncommon Breed :star: :star: :star:
Flavor: It's another variation on the "wild encounter but stronger" but the twist fails to land as well because it's not unlockable
Gameplay: Not unlocking the egg moves permanently is criminal. One of the most important things in the game should be allowing more metaprogression unlocks so arbitrarily not allowing the permanent unlock is weird. If the move unlocks were permanent this could have easily been a 5 star event 
Code: Since this shares the base idea with berries abound, it makes me think there should be something in ME utils for better scalability of "wild encounter but with a twist"

**Comment by torranx** [comment]
might want to move this to https://github.com/Despair-Games/poketernity/discussions ?
